### PR TITLE
tkn create Command Proposal

### PR DESCRIPTION
**Pull request by danielhelfand**
Associated with #574 

This pull request is a proposal for a `tkn create` command that invokes `kubectl create`. It would work as follows:

```
tkn create -f foo.yaml -n bar
```

This closely mirrors the approach implemented in [`ko delete`](https://github.com/google/ko/blob/1fcfe62a9606a2781559ed10897a4ae7141d7628/pkg/commands/delete.go). 

Pros of this approach:
* Less tool switching for users (Users will have to use `kubectl` less frequently when working with Tekton)
* Ability to create non Tekton resources that are used with Tekton

Cons:
* Requires `kubectl` to be installed/dependence on `kubectl` (could possibly swap out kubectl commands with this)
* We will most likely need to rely on `e2e` testing to support this without unit tests

# Submitter Checklist

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Regenerate the manpages and docs with `make docs` and `make man` if needed.
- [x] Run the code checkers with `make check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```
Add tkn create command that invokes kubectl create
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -414,6 +414,7 @@ k8s.io/apimachinery v0.0.0-20191004074956-01f8b7d1121a h1:lDydUqHrbL/1l5ZQrqD1RI
 k8s.io/apimachinery v0.0.0-20191004074956-01f8b7d1121a/go.mod h1:ccL7Eh7zubPUSh9A3USN90/OzHNSVN6zxzde07TDCL0=
 k8s.io/cli-runtime v0.0.0-20191004110054-fe9b9282443f h1:vJOrMsZe+RD884n+WQ5So2oOp7SajI0Op3oOBg64ZsY=
 k8s.io/cli-runtime v0.0.0-20191004110054-fe9b9282443f/go.mod h1:qWnH3/b8sp/l7EvlDh7ulDU3UWA4P4N1NFbEEP791tM=
+k8s.io/cli-runtime v0.17.0 h1:XEuStbJBHCQlEKFyTQmceDKEWOSYHZkcYWKp3SsQ9Hk=
```
Comment (chmouel) [inline go.sum]: I don't think you wanted this?

File: pkg/cmd/create/create.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			UnknownFlags: true,
+		},
+	}
+
+	return c
+}
+
+// returns a runCmd that simply passes CLI arguments
+// through to a binary named command.
+func invokeKubectl(command string) runCmd {
```
Comment (chmouel) [inline pkg/cmd/create/create.go]: What do you think about adding a check to look out for the binary and let the user that she would need to install `kubectl` before executing this command?

File: go.sum (hunk 1 above)
Comment (vdemeester) [reply go.sum]: @chmouel it's on the `sum` file, doesn't mean we get the dependency :stuck_out_tongue:

File: pkg/cmd/create/create.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	c := &cobra.Command{
+		Use:          "create",
+		Short:        "Uses kubectl create to create Kubernetes resources",
+		Example:      eg,
+		SilenceUsage: true,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Run: invokeKubectl("kubectl"),
```
Comment (vdemeester) [inline pkg/cmd/create/create.go]: shouldn't it be `invokeCommand` then ? :stuck_out_tongue_closed_eyes: (as with the function itself doesn't have anything to do with `kubectl`)

Comment (danielhelfand) [reply pkg/cmd/create/create.go]: Yeah, that's a good call. I'll touch it up.

File: pkg/cmd/create/create.go (hunk 2 above)
Comment (danielhelfand) [reply pkg/cmd/create/create.go]: Agreed. Let me add a check for `kubectl` binary.

Comment (danielhelfand) [reply pkg/cmd/create/create.go]: Check added via `isKubectlAvailable()`

**Comment by chmouel** [comment]
/lgtm
/approve

Thanks 👍🏼🔥

**Pull request merged**